controllers/teleport_controller/teleport_controller.cpp

```diff
@@ -0,0 +1,138 @@
+// File:          teleport_controller.cpp
+// Date:          23/03/2021
+// Description:   Method to teleport robot around the playing field for the purposes of
+//                    gathering training data
+// Author:        Lachlan Court
+// Modifications:
+
+
+#include <RobotisOp2GaitManager.hpp>
+#include <RobotisOp2MotionManager.hpp>
+#include <webots/Supervisor.hpp>
+#include <iostream>
+#include <math.h>
+#include <time.h> 
+
+// All the webots classes are defined in the "webots" namespace
+using namespace webots;
+
+/*
+// Rotations matrix for 90 degree rotations
+static const int rotationsSize = 4;
+static const double rotations[rotationsSize][4] = {{1, 0, 0, 1.57081},
+                                                  {0.577353, 0.577345, 0.577353, 2.0944},
+                                                  {1.69526e-09, 0.707103, 0.707111, 3.14159},
+                                                  {-0.577352, 0.577347, 0.577352, -2.09441}};
+
+*/
+
+// Rotations matrix for 45 degree rotations
+static const int rotationsSize = 8;                                     
+static const double rotations[rotationsSize][4] = {{1, 0, 0, 1.57081},
+                                                  {0.867737, 0.343283, 0.359429, 1.75753},
+                                                  {0.577353, 0.577345, 0.577353, 2.0944},
+                                                  {0.286947, 0.661632, 0.692752, 2.60662},
+                                                  {1.69526e-09, 0.707103, 0.707111, 3.14159},
+                                                  {-0.286948, 0.661631, 0.692752, -2.60661},
+                                                  {-0.577352, 0.577347, 0.577352, -2.09441},
+                                                  {0.867739, -0.343277, -0.359429, 1.75753}};                               
+
+
+/* This is the main program of the supervisor. It creates an instance of the Supervisor
+ *     instance, launches its function(s) and destroys it at the end of the execution.
+ */
+int main() {
+    std::cout << "a" << std::endl;
+    // create the Supervisor instance and assign it to a robot
+    Supervisor* supervisor = new Supervisor();
```

Comment:
Never use raw pointers (`new` and `delete`). Use smart pointers instead, or more conveniently, just do everything on the stack. Our codebase doesn't do much stuff with objects large enough to bother with heap allocations, so most of the time we just do everything on the stack, with the added bonus of performance.